dmutils/repoutils/freeze-requirements.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+"""
+Output a file containing the full, frozen dependencies of a requirements file.
+
+This will error out on the pip
+    > Requirement file x contains y, but that package is not installed...
+error caused by `pip install` being case insensitive and the `pip freeze` command being case sensitive.
+
+Usage:
+    freeze-requirments <target>
+    freeze-requirments <target> [options]
+
+Options:
+    <target>             A requirements file for us to freeze
+    --outfile=<outfile>  Output file for frozen requirement values [default: requirements.txt]
+    --verbose
+    -h, --help           Show this screen
+
+Examples:
+    python -m dmutils.repoutils.freeze-requirments requirements-app.txt
```

Comment:
I just tried running this locally (against the supplier FE, in the virtualenv) and it seems fine. However the current `make freeze-requirements` is done outside of the supplier FE virtualenv. The `make freeze-requirements` Makefile command have to be something like `${VIRTUALENV_ROOT}/bin/python -m [etc]`, so we should probably update this example. (What happens if we're bootstrapping and haven't installed utils yet? Does this ever happen in practice?)

Also there are a few typos `freeze-requirments` 😸

Reply:
Doh, typos, will correct.

Looks like the current `make freeze-requirements` will operate inside or outside of the venv using whatever python interpreter it finds in `echo $PATH`.
But you're right, in the case of our repos this _shouldn't_ be run outside of the venv because we want to ensure we're using the right utils version.

If we specify that here we'd have to say something like 'You can't use this module without installing it' which is true for any module.

What we do in other cases, like for `flake8` and `py.test` is force this in the makefile like so:

```
.PHONY: test-flake8
test-flake8: virtualenv
	${VIRTUALENV_ROOT}/bin/flake8 .
```

So we could do:
```
.PHONY: freeze-requirements
freeze-requirements: virtualenv
	${VIRTUALENV_ROOT}/bin/python -m dmutils.repoutils.freeze-requirements requirements-app.txt
```

Reply:
typos done 👍

Reply:
(ie there should always be a venv available)